src/dbnode/network/server/httpjson/handlers.go

```diff
@@ -206,6 +228,7 @@ func RegisterHandlers(mux *http.ServeMux, service interface{}, opts ServerOption
 				return
 			}
 
+			w.WriteHeader(http.StatusOK)
```

Comment:
This may be redundant. From docs: https://golang.org/pkg/net/http/#ResponseWriter
```
        // If WriteHeader has not yet been called, Write calls
        // WriteHeader(http.StatusOK) before writing the data.
```
Not sure if that could cause the issues we're debugging

Reply:
Update: I removed this line and saw the same EOF error, not the issue